mm2src/coins/lp_coins.rs

```diff
@@ -3511,3 +3520,27 @@ where
         b.block_height.cmp(&a.block_height)
     }
 }
+
+#[derive(Display, Debug)]
+pub enum RpcCommonError {
+    FindClientError(String),
+    WrongRpcClient,
+}
+
+#[derive(Debug)]
+pub enum RpcClientEnum {
+    #[cfg(not(target_arch = "wasm32"))]
+    ZcoinRpcClient(CompactTxStreamerClient<TonicChannel>),
+    TendermintHttpClient(HttpClient),
+    Web3HttpTransportNode(HttpTransportNode),
+}
+
+/// Use trait in the case, when we have to send requests to rpc client.
+#[async_trait]
+pub trait RpcCommonOps {
+    /// get alive client from custom `RpcClient` structure, that contains one rpc client and vector of rpc urls.
+    async fn get_rpc_client(&self) -> Result<RpcClientEnum, RpcCommonError>;
+
+    /// if rpc client isn't available anymore, iterate over rpc urls, to get the first available client.
+    async fn iterate_over_urls(&self) -> Result<RpcClientEnum, RpcCommonError>;
```

Comment:
I think `get_working_client` or `get_live_client` can be a better name for this functions. What do you think @laruh?

Reply:
Yes, I agree. This is our responsibility to be sure in this function, that current client or new one are alive, so I will use `get_live_client`. will add I with other changes.

Reply:
done